### DISCUSSION TITLE
IsDisplayed() shows as unresolved method, although still works

### DISCUSSION THREAD

**Discussion by annastrik** [webdriverio/webdriverio#11708] [Q&A]
Hi WebdriverIO community,
Can someone help me to figure out why the isDisplayed() method shows as greyed out for me, not enabled, not clickable. When I hover over it shows “unresolved function or method”.

However, the isDisplayed function still works normally, I checked in logs.

Same concerns waitForExist() and waitForDisplayed(). It also works by the way, just shows as “unresolved function or method”.

I use Appium/WebdriverIO/Mocha integration in my tests.

![Screenshot 2023-11-21 at 1 06 54 PM](https://github.com/webdriverio/webdriverio/assets/58706349/fdeeed82-2b66-4baa-b8b6-3804d1e9ad68)
<img width="876" alt="Screenshot 2023-11-21 at 1 06 46 PM" src="https://github.com/webdriverio/webdriverio/assets/58706349/471d51bb-2e1c-4fde-9d7a-2332d4b34436">
![Screenshot 2023-11-21 at 1 13 56 PM](https://github.com/webdriverio/webdriverio/assets/58706349/26b15657-0be0-4a77-a490-62caf3d9465e)

Package.json:
{
  "name": "xxx",
  "version": "1.0.0",
  "description": "xxx",
  "author": "xxx",
  "license": "xxx",
  "scripts": {
    xxx},
  "engines": {
    "node": ">=10",
    "npm": ">=6"
  },
  "devDependencies": {
    "@babel/core": "^7.21.5",
    "@babel/preset-env": "^7.21.5",
    "@babel/register": "^7.21.0",
    "allure-commandline": "^2.22.0",
    "allure-mocha": "^2.2.1",
    "appium": "2.0.1",
    "appium-uiautomator2-driver": "2.29.3",
    "appium-xcuitest-driver": "4.33.0",
    "axios": "^1.4.0",
    "chai": "^4.3.7",
    "eslint": "^8.39.0",
    "eslint-plugin-mocha": "^10.1.0",
    "lodash": "4.17.21",
    "mocha": "10.2.0",
    "mocha-allure-reporter": "1.4.0",
    "mocha-junit-reporter": "2.0.0",
    "testcafe-reporter-testrail": "0.6.3"
  },
  "dependencies": {
    "form-data": "^4.0.0",
    "protobufjs": "^6.11.2",
    "uuid": "9.0.1",
    "webdriverio": "8.23.4"
  }
}

**Comment by christian-bromann** [comment]
@annastrik I double checked the release and both command types and implementation got properly released. We recently did some changes here so I would not be suprised if we missed something. Would you mind providing a reproducible example?

**Reply by annastrik** [reply]
@christian-bromann 
Hi Christian, thank you for your answer!
What do you mean under reproducible example? It is my company repository..

I create a session using this function:
return await wdio.remote(capabilities)
With the following capabilities:

let capabilities = {
      host: '127.0.0.1',
      port:4724, 
      path:'/',
      logPath:'./artifacts/appiumServerLog.log',
      logLevel:'error',
      localServer:'enabled',
      showXcodeLog:true,
      capabilities: {
         platformName:'iOS',
         'appium:automationName':'XCUITest',
         'appium:deviceName':'iPhone 12',
         'appium:platformVersion':'16.2',
         'appium:app':'/Users/xxx/xxxx/artifacts/xxx.app,
         'appium:appPackage':'xxx',
         'appium:orientation':'PORTRAIT',
         'appium:showIOSLog':false,
         'appium:language':'en',
         'appium:noReset':false,
         'appium:autoAcceptAlerts':true,
         'appium:newCommandTimeout':0
      }
   };


What I noticed is that when I add “wd” in dependencies, the isDisplayed() becomes active:

 "dependencies": {
   "form-data": "4.0.0",
   "protobufjs": "7.2.5",
   "uuid": "9.0.1",
   "wd": "1.14.0",
   "webdriverio": "8.23.4"
 }
}
![Screenshot 2023-11-22 at 9 57 33 AM](https://github.com/webdriverio/webdriverio/assets/58706349/6686ec85-03f9-4c92-bde1-e3ba6365f048)

Does it mean that isDisplayed() function is taken from “wd” and not from webdriverio?

The tests work the same in both cases (when wd is in dependencies and isDisplayed() is active and when “wd” is not in dependencies and isDisplayed() is not active) - the logs are absolutely the same. The only problem is that isDisplayed() is not active and not clickable. I am wondering why is it like that?

Sorry, I am not technically advanced.

Here are the logs:

[HTTP] --> GET /session/b62c62a2-512c-4436-a4ce-5aa80bd9cd11/element/2E000000-0000-0000-8C05-000000000000/displayed
[HTTP] {}
[XCUITestDriver@6a59 (b62c62a2)] Driver proxy active, passing request on via HTTP proxy
[debug] [XCUITestDriver@6a59 (b62c62a2)] Executing command 'proxyReqRes'
[debug] [XCUITestDriver@6a59 (b62c62a2)] Matched '/session/b62c62a2-512c-4436-a4ce-5aa80bd9cd11/element/2E000000-0000-0000-8C05-000000000000/displayed' to command name 'elementDisplayed'
[debug] [XCUITestDriver@6a59 (b62c62a2)] Proxying [GET /session/b62c62a2-512c-4436-a4ce-5aa80bd9cd11/element/2E000000-0000-0000-8C05-000000000000/displayed] to [GET http://127.0.0.1:8100/session/2FABDF59-A1B9-4182-BBBA-0108E131254A/element/2E000000-0000-0000-8C05-000000000000/displayed] with no body
[debug] [XCUITestDriver@6a59 (b62c62a2)] Got response with status 200: {"value":true,"sessionId":"2FABDF59-A1B9-4182-BBBA-0108E131254A"}
[XCUITestDriver@6a59 (b62c62a2)] Replacing sessionId 2FABDF59-A1B9-4182-BBBA-0108E131254A with b62c62a2-512c-4436-a4ce-5aa80bd9cd11
[HTTP] <-- GET /session/b62c62a2-512c-4436-a4ce-5aa80bd9cd11/element/2E000000-0000-0000-8C05-000000000000/displayed 200 105 ms - 65

**Reply by christian-bromann** [reply]
If you use WebdriverIO you should not use `wd`, remove it from your dependencies.

**Reply by annastrik** [reply]
Yes, I know. That is what I did initially. There was no “wd” from the start.
When I have this in package.json (no “wd”) the isDisplayed() is disabled:

 dependencies": {
"form-data": "^4.0.0",
"protobufjs": "^6.11.2",
"uuid": "9.0.1",
"webdriverio": "8.23.4"
}
}

**Reply by erwinheitzman** [reply]
What we mean by a reproducible example repo is that you create a small repository in which you install webdriverio and appium in which you can reproduce the issue.
Without it, we are simply playing in the dark.

**Reply by annastrik** [reply]
@erwinheitzman @christian-bromann 
Thank you for clarifying, here it is
https://github.com/annastrik/Project-Appium-Mocha-WDIO-Sample

Just open any of 2 files in tests directory.
The way I see isDisplayed, waitForDisplayed, waitForExist in Webstorm:

<img width="854" alt="Screenshot 2023-11-27 at 6 58 47 PM" src="https://github.com/webdriverio/webdriverio/assets/58706349/d9170ed2-e5d6-413c-98f6-0496879873d8">

**Reply by erwinheitzman** [reply]
Looks like it's all the commands as the click command is also highlighted so I doubt this has got anything to do with the changes to the isDisplayed command

**Reply by christian-bromann** [reply]
I don't have a Webstorm license so it is hard for me to debug this. Any help is much appreciated.

**Reply by annastrik** [reply]
@erwinheitzman the click command had some eslint warning (duplicated code note), it is not related, I ignored it.

Here are the texts that appear when hovering over isDisplayed and waitForDisplayed (it is not the eslint warning):
![Screenshot 2023-11-28 at 10 39 58 AM](https://github.com/webdriverio/webdriverio/assets/58706349/1753a947-5cf6-4c7f-8a6e-41b4fe2228e0)
![Screenshot 2023-11-28 at 10 39 55 AM](https://github.com/webdriverio/webdriverio/assets/58706349/6c161c09-1341-4f9e-a16c-578b1682446f)
